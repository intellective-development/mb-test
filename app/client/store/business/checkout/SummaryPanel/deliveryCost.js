import React from 'react';
import { css } from '@amory/style/umd/style';
import styles from '../Checkout.css.json';

// @flow
export const deliveryCost = (shoprunner: boolean, fallback: string | JSX.Element, containerClass?: string = 'shoprunner__container') => (
  shoprunner ? (
    <div className={`_SRD ${css([styles[containerClass]])}`}>
      <div className={`srd_iconline ${css([styles.shoprunner])}`}>
        <div className="srd_logo" />
      </div>
    </div>
  ) : fallback
);

export default deliveryCost;
