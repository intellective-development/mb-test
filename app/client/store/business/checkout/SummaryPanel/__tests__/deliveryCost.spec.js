
import deliveryCost from '../deliveryCost';

describe('deliveryCost', () => {
  it('renders an icon when ShopRunner is available', () => {
    expect(deliveryCost(true, null)).toMatchSnapshot();
  });

  it('renders an icon with specific styling when ShopRunner is available', () => {
    expect(deliveryCost(true, null, 'shoprunner__container_inline')).toMatchSnapshot();
  });

  it('renders fallback cost when ShopRunner is not available', () => {
    expect(deliveryCost(false, 'fallback')).toMatchSnapshot();
  });
});
