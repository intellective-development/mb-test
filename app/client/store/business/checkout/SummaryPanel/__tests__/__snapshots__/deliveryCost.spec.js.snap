// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deliveryCost renders an icon when ShopRunner is available 1`] = `
<div
  className="_SRD crbno c4zh6 e8gsh"
>
  <div
    className="srd_iconline dreop eg0o5"
  >
    <div
      className="srd_logo"
    />
  </div>
</div>
`;

exports[`deliveryCost renders an icon with specific styling when ShopRunner is available 1`] = `
<div
  className="_SRD dreop e6wjf egvfd"
>
  <div
    className="srd_iconline dreop eg0o5"
  >
    <div
      className="srd_logo"
    />
  </div>
</div>
`;

exports[`deliveryCost renders fallback cost when ShopRunner is not available 1`] = `"fallback"`;
